otel-extension/src/it/scala/io/scalac/mesmer/agent/utils/OtelAgentTest.scala

```diff
@@ -0,0 +1,21 @@
+package io.scalac.mesmer.agent.utils
+
+import io.opentelemetry.context.ContextStorage
+import io.opentelemetry.instrumentation.testing.AgentTestRunner
+import io.opentelemetry.instrumentation.testing.util.ContextStorageCloser
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.BeforeAndAfterEach
+import org.scalatest.TestSuite
+
+abstract class OtelAgentTest extends TestSuite with BeforeAndAfterAll with BeforeAndAfterEach {
+
+  private val testRunner = AgentTestRunner.instance()
```

Comment:
I sense that this is the most crucial part of this PR, right? It still should run a test suit per process? Previously we limited amount of test run in the separate process to allow any suite to register it's transformations and not intervene with previously registered - is it still the case?